src/components/progressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useEffect } from "react";
+
+// Define the prop types for the ProgressBar component
+type ProgressBarProps = {
+  currentValue: number;
+  maxValue: number;
+  backgroundColor?: string;
+  progressColor?: string;
+  isLinear?: boolean;
+  displayPercentage?: undefined | number | string;
+};
+
+const ProgressBar: React.FC<ProgressBarProps> = ({
+  currentValue = 0,
+  maxValue = 100,
+  backgroundColor = "bg-progress-bg",
+  isLinear = false,
+  displayPercentage,
+  progressColor = "bg-progress-bar",
+}) => {
+  // State to manage the percentage
+  const [percentage, setPercentage] = useState((currentValue / maxValue) * 100);
+
+  // Inline style for the progress bar
+  const barStyle = {
+    width: isLinear ? `${percentage}%` : "100%",
+    backgroundColor: progressColor,
+  };
+
+  return isLinear ? (
+    <LinearProgressBar
```

Comment:
these should take an additional prop (percentage) which is computed above.

I think framer supports animations to run on initialization. We should be able to pass that along and use framer to create an animation on initial load and any subsequent changes to the above values.